### PR TITLE
Option of excluding sub pattern types from the styleguide

### PR DESCRIPTION
**Pull request by AndrewLeedham**
Addresses: Option of excluding sub pattern types from the styleguide

Summary of changes:
* Replicated omitPatternType for 'patternType/patternSubtype'
* Duplicated typePatterns as styleguideTypePatterns to seperate stylguide and nav menu creation.

### CODE REVIEW

File: core/lib/ui_builder.js [hunk 1]
```diff
@@ -508,6 +508,19 @@ var ui_builder = function () {
...
         });
 
+        //determine if we should omit this subpatterntype completely from the viewall page
+        var omitPatternType = styleGuideExcludes && styleGuideExcludes.length
+          && _.some(styleGuideExcludes, function (exclude) {
+            return exclude === patternType + '/' + patternSubtype;
+          });
+        if (omitPatternType) {
+          if (patternlab.config.debug) {
+            console.log('Omitting ' + patternType + '/' + patternSubtype + ' from  building a viewall page because its patternGroup is specified in styleguideExcludes.');
```
Comment (bmuenzenmeyer) [inline core/lib/ui_builder.js]: Should this be pattern subGroup?

Comment (AndrewLeedham) [inline core/lib/ui_builder.js]: You're not wrong, must of missed that, I'll commit the change now

**Comment by AndrewLeedham** [comment]
@bmuenzenmeyer Thanks for the feedback and quick reply. With regards to usage examples, what do you mean by a snippet? The intention is that sub-types can be specified within the already existing 'styleGuideExcludes' array with the syntax "patternType/subPatternType". Since sub-types are user defined, adding this to the default `patternlab-config.json` wouldn't make all too much sense.

**Comment by bmuenzenmeyer** [comment]
By snippet I meant how would one exclude a subtype.

```
...
"styleguideExcludes" : ["atoms/global"]
...
```

would do the trick if I understand correctly

**Comment by AndrewLeedham** [comment]
@bmuenzenmeyer Sorry I worded that pretty badly. I meant where do you want me to include the snippet? in the default patternlab-config.json?

**Comment by bmuenzenmeyer** [comment]
just here, so I can make sure to mention it in the release notes or wiki 
if i have that syntax as you'd expect, we are all good 🚀

**Comment by AndrewLeedham** [comment]
@bmuenzenmeyer Ah okay fair enough. Yeah you have it right.

**Pull request merged**